x-pack/plugins/security_solution/public/common/store/store.ts

```diff
@@ -166,6 +164,90 @@ export const createStoreFactory = async (
   ]);
 };
 
+const timelineActionsWithNonserializablePayloads = [
+  timelineActions.updateTimeline.type,
+  timelineActions.addTimeline.type,
+  timelineActions.updateAutoSaveMsg.type,
+  timelineActions.initializeTimelineSettings.type,
+];
+
+const actionSanitizer = (action: AnyAction) => {
+  if (action.type === sourcererActions.setDataView.type) {
+    return {
+      ...action,
+      payload: {
+        ...action.payload,
+        dataView: 'dataView',
+        browserFields: 'browserFields',
+        indexFields: 'indexFields',
+        fields: 'fields',
+      },
+    };
+  } else if (timelineActionsWithNonserializablePayloads.includes(action.type)) {
```

Comment:
Switching `timelineActionsWithNonserializablePayloads` to an object like this: 

```js
const timelineActionsWithNonserializablePayloads = {
  [timelineActions.updateTimeline.type]: 1,
  [timelineActions.addTimeline.type]: 1,
  [timelineActions.updateAutoSaveMsg.type]: 1,
  [timelineActions.initializeTimelineSettings.type]: 1,
};
```

to make the lookup ever so slightly faster:

```js
if (timelineActionsWithNonserializablePayloads[action.type]) {...}
```